documentation/compatibility_with_puppet_agent.markdown

```diff
@@ -1,106 +1,100 @@
-have no effect on typical deployments.
-
-# Configuration settings
-
-When operating Puppet 3 agents against Puppet Server 2.1, some configuration
-changes on the agents may be required ahead of time.  Two settings in
-particular should be configured:
-
- * `stringify_facts = false` which is required with a Puppet 4 master.
- * The master will operate with the future parser turned on, so if the future
-   parser is not being used with Puppet 3, please turn it on and try it out
-   before upgrading to Puppet Server 2.1.  Any incompatibilities between
-   existing Puppet code and the future parser should be resolved prior to
-   upgrading to Puppet Server 2.1.
-
-# auth.conf
-
-The REST API URL structure has changed starting in Puppet 4.  The major change
-is the introduction of a versioned API rooted at `/puppet`, with the CA API
-separated out at `/puppet-ca`.  In the general case a Puppet 3 URL can be
-translated to Puppet 4 by stripping off the environment name prefix, pre-pending
-`/puppet/v3`, and then adding back the environment name as a query parameter.
-If `auth.conf` has been customized and Puppet Server 2.1 is going to be used
-with Puppet 3 agents then the Puppet 3 requests need to be translated in this
-way and expressed in auth.conf in their Puppet 4 form.
-
-For example, consider a customized deployment that uses UUID's to uniquely
-identify certificates, but omits the UUID from the node-name to facilitate
-de-provisioning and re-provisioning the same node identity using unique
-certificates.
-
-    # puppet.conf on the agent
-    [main]
-    certname = emanon.uuid.ec7f5196-7f63-5f73-f18d-ca69afc5c24d
-    node_name_value = emanon.uuid
-
-To support this configuration in Puppet 3 auth.conf has been customized as
-follows:
+---
+layout: default
+title: "Puppet Server: Backwards Compatibility With Puppet 3 Agents"
+canonical: "/puppetserver/latest/compatibility_with_puppet_agent.markdown"
+---
+
+
+[ca.conf]: ./configuration.html#caconf
+[auth.conf]: /puppet/latest/reference/config_file_auth.html
+
+This version of Puppet Server can serve configurations to both Puppet 4 and Puppet 3 agent nodes. This feature was added in Puppet Server 2.1.
+
+Once you get your Puppet 3 nodes talking to a newer Puppet Server, you should start upgrading them to Puppet 4.
+
+
+## Preparing Puppet 3 Nodes for Puppet Server 2
+
+Backwards compatibility is enabled by default, but you should make sure your Puppet code is ready for Puppet 4 before pointing old agent nodes at a new Puppet server. You may also need to edit auth.conf.
+
+Before migrating Puppet 3 nodes to Puppet Server 2, do all of the following:
+
+- Update Puppet to the most recent 3.x version on your agent nodes and existing Puppet master.
+- Set `stringify_facts = false` on all agent nodes, and fix your code if necessary.
+- Make sure you're already running with the future parser enabled, and fix your code if necessary.
```

Comment:
And I spoke too soon.. loos like it's directly below. Hmm maybe just add "see instructions below" or something?